server/src/main/java/org/opensearch/repositories/blobstore/ChecksumBlobStoreFormat.java

```diff
@@ -222,7 +222,7 @@ public void writeAsync(
                     WritePriority.HIGH,
                     (size, position) -> new OffsetRangeIndexInputStream(input, size, position),
                     expectedChecksum,
-                    true
+                    ((AsyncMultiStreamBlobContainer) blobContainer).remoteIntegrityCheckSupported()
```

Comment:
not all `BlobContainer` would be of type `AsyncMultiStreamBlobContainer` . Hence we would need to check the type here and then decide accordingly.

Reply:
we already check for it above

Reply:
````
if (blobContainer instanceof AsyncMultiStreamBlobContainer == false) {
            write(obj, blobContainer, name, compressor);
            listener.onResponse(null);
            return;
        }
```
RemoteTransferManager call is not being made if it is not the instance type.